src/experimental/tools/ContentDownloader/api/downloader/downloadManager.ts

```diff
@@ -0,0 +1,111 @@
+/**
+ * Copyright 2019 CANAL+ Group
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { AsyncSubject, combineLatest, of } from "rxjs";
+import { filter, startWith } from "rxjs/operators";
+
+import { SegmentPipelineCreator } from "../../../../../core/pipelines";
+import { ICallbacks, IInitSettings, IStoredManifest } from "../../types";
+import { initDownloader$ } from "./initSegment";
+import { getTransportPipelineByTransport } from "./manifest";
+import { segmentPipelineDownloader$ } from "./segment";
+import { IDownloadManagerOptions } from "./types";
+
+/**
+ * DownloadManager that will handle actions to take depending if we are in
+ * resuming or downloading from scratch.
+ */
+class DownloadManager {
+  readonly downloadManagerOptions: IDownloadManagerOptions;
+
+  constructor(downloadManagerOptions: IDownloadManagerOptions) {
+    this.downloadManagerOptions = downloadManagerOptions;
+  }
+
+  initDownload(initSettings: IInitSettings, pause$: AsyncSubject<void>) {
+    const { contentID, onError, onProgress } = initSettings;
+    const builderInit = {
+      progress: { percentage: 0, segmentsDownloaded: 0, totalSegments: 0 },
+      manifest: null,
+      video: [],
+      audio: [],
+      text: [],
+      size: 0,
+    };
+    const pipelineSegmentDownloader$ = segmentPipelineDownloader$(
+      initDownloader$(initSettings, this.downloadManagerOptions.db),
+      builderInit,
+      { contentID, onError, onProgress, db: this.downloadManagerOptions.db, pause$ }
+    );
+    return combineLatest([
+      pipelineSegmentDownloader$.pipe(
+        // TODO: See what we can do here with, this define the frequency of save
+        filter(({ progress: { percentage } }) => percentage % 10 === 0),
```

Comment:
Indeed, we could jump over the `% 10 === 0`, to correct that we know `round` the `percentage` and then apply a `distinct` operator to avoid sending as event the same `percentage`.

Concerning your second writing, what do you mean by `I see not mention of that` ?